### PR TITLE
e2e: Proposal Create mock structure. 

### LINKED ISSUE #2644: e2e: Proposal Create mock structure.
Part of #2642.

Proposal Create tests are still using the online api approach. We need to migrate it to the new mock structure introduced by (#2624).

We currently don't need new test suites, the main focus here is to first migrate and then code more tests using this new structure.

## Tips and Tricks

- Only mock requests related to Proposal Create (`record/new`) page
- Run only `create` test file: `yarn cy:run -- --spec "cypress/e2e/proposal/create.js"`
- Setup the API environment for all possible routes reached by the tests suite using a `beforeEach` statement
- Check on politeiawww's logs to check if all requests were mocked properly
- Use `proposal/list.js` tests as reference

### PR DESCRIPTION
**Pull request by vibros68**
Closes #2644

This diff create mock data for `record/new` endpoint. So when creating
proposal e2e test is running. The test will not send the request to the 
server. And complete the e2e test for creating normal proposal. Include 
select domain and pickup datepickers

### CODE REVIEW

**Comment by vibros68** [comment]
@victorgcramos can you review this pr?

**Review by victorgcramos — CHANGES_REQUESTED** [review]
`records/v1/new` endpoint mock looks good, mate. Thanks! Left some inline comments.

Good to remind that mocking the records/new endpoint does not close the full issue, as It expected the whole proposal/create flow to be mocked so we don't need to be connected to the backend.

File: teste2e/cypress/support/commands.js [hunk 1]
```diff
@@ -149,22 +158,24 @@ Cypress.Commands.add("approveProposal", ({ token }) =>
 Cypress.Commands.add("typeCreateProposal", (proposal) => {
   cy.server();
   cy.findByTestId("proposal-name").type(proposal.name);
-  cy.findByTestId("text-area").type(proposal.description);
+  cy.findByTestId("proposal-amount").type(String(proposal.amount / 100)); // get dollars from cents.
```
Comment (victorgcramos) [inline teste2e/cypress/support/commands.js:161]: Prettier: 80 max char per line.

File: teste2e/cypress/support/core/api.js [hunk 2]
```diff
@@ -101,7 +101,32 @@ export function detailsReply({
   return { record };
 }
 
+/**
+ * newRecordReply is the reply to the new command. It returns a new record for the
+ * request data with given `files`, `publickey`, `signature` and `author` testParams.
```
Comment (victorgcramos) [inline teste2e/cypress/support/core/api.js:106]: The parameters from this replier doc is different from the replier's parameters.

File: teste2e/cypress/support/core/api.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Record, File, Inventory } from "./generate";
+import { Record, File, Inventory, Proposal } from "./generate";
```
Comment (victorgcramos) [inline teste2e/cypress/support/core/api.js:1]: `Proposal` is never used and does not exist on generate scope.

File: teste2e/cypress/e2e/proposal/create.js [hunk 4]
```diff
@@ -1,41 +1,69 @@
...
+  cy.useCommentsApi();
+});
+
 describe("Proposal Create", () => {
   // XXX This test needs changes in the Datepicker and (probably) the Select
   // components, in order to fill the new form fields such as: start & end dates
   // and amount - issue to track <insert issue link>
   //
-  /*it("Paid user can create proposals manually", () => {
+  it("Paid user can create proposals manually", () => {
```
Comment (victorgcramos) [inline teste2e/cypress/e2e/proposal/create.js:18]:
```suggestion
  it("should be able to create proposals", () => {
```

**Review by victorgcramos — CHANGES_REQUESTED** [review]
Looking better now. Tests are working as expected, and thanks for adding an extra test case.

I have some small observations regarding the test cases descriptions: they should start with "should". Left one inline suggestion regarding it, which you should replicate for all test cases you create.

Left some other inline suggestions 😁

File: teste2e/cypress/support/commands.js [hunk 5]
```diff
@@ -25,13 +25,22 @@
 // Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
 //
 import { sha3_256 } from "js-sha3";
-import { requestWithCsrfToken, setProposalStatus } from "../utils";
+import {
+  PROPOSAL_SUMMARY_STATUS_UNVETTED,
```
Comment (victorgcramos) [inline teste2e/cypress/support/commands.js:29]: Not being used.

File: teste2e/cypress/support/commands.js [hunk 6]
```diff
@@ -25,13 +25,22 @@
...
 // TODO: consider moving general functions like makeProposal and signRegister
 // to a more general lib file other than api.
-import { makeProposal, signRegister } from "../utils";
-import { shortRecordToken } from "../utils";
+import {
+  makeProposal,
+  signRegister,
+  shortRecordToken,
+  RECORD_DOMAINS,
+  typeDatePicker
```
Comment (victorgcramos) [inline teste2e/cypress/support/commands.js:42]: Not being used.

**Review by victorgcramos — CHANGES_REQUESTED** [review]
I'm getting an error on your test run. Please fix it :D

Screen Shot 2021-12-02 at 12 15 34 PM

File: teste2e/cypress/support/core/generate.js [hunk 7]
```diff
@@ -106,15 +108,19 @@ export function Record({
...
   this.metadata = [
     UserMetadata(user),
     RecordMetadata({
       token,
       version,
       status,
-      publickey: user.publickey,
-      signature,
+      publickey: publickey || user.publickey,
+      recordSignature,
```
Comment (victorgcramos) [inline teste2e/cypress/support/core/generate.js:123]: Noticed some issues here. RecordMetadata has no `recordSignature` field. Actually, it's `signature`. Also, the signature should have a default value.

**Review by victorgcramos — CHANGES_REQUESTED** [review]
Looking really good. Left just one minor inline comment, and we should be good to go! Nice job @vibros68

File: teste2e/cypress/e2e/proposal/create.js [hunk 8]
```diff
@@ -15,7 +15,7 @@ describe("Proposal Create", () => {
   // components, in order to fill the new form fields such as: start & end dates
   // and amount - issue to track <insert issue link>
   //
-  it("Paid user can create proposals manually", () => {
+  it("should be able to create proposals", () => {
```
Comment (amass01) [inline teste2e/cypress/e2e/proposal/create.js:18]:
```suggestion
  it("should allow paid user to create proposals", () => {
```

**Review by victorgcramos — APPROVED** [review]
tACK!

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM

**Pull request merged**